WalletWasabi/Helpers/StartupHelper.cs

```diff
@@ -0,0 +1,77 @@
+using Microsoft.Win32;
+using System;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using WalletWasabi.Logging;
+
+namespace WalletWasabi.Helpers
+{
+	public static class StartupHelper
+	{
+		public const string StartupErrorMessage = "Something went wrong while trying to make your changes.";
+
+		public static bool TryModifyStartupSetting(bool isWasabiStartsWithOS)
+		{
+			if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+			{
+				return TryModifyRegistry(isWasabiStartsWithOS);
+			}
+			else if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+			{
+				// Method call here
+			}
+			else if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+			{
+				// Method call here
+			}
+			return false;
+		}
+
+		private static bool TryModifyRegistry(bool isWasabiStartsWithOS)
+		{
+			string keyName = "SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Run";
+			try
+			{
+				// This extra check is here only to eliminate warnings.
+				if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+				{
+					throw new InvalidOperationException("It can only be called on Windows.");
+				}
+
+				using RegistryKey key = Registry.CurrentUser.OpenSubKey(keyName, writable: true) ?? throw new NullReferenceException();
+				if (isWasabiStartsWithOS)
+				{
+					string pathToExeFile = Assembly.GetExecutingAssembly().Location;
+					pathToExeFile = pathToExeFile.Remove(pathToExeFile.Length - 4);        // This part has to change if this gets released
+					pathToExeFile += ".Fluent.Desktop.exe";
+
+					key.SetValue("WasabiWallet", pathToExeFile);
+				}
+				else
+				{
+					key.DeleteValue("WasabiWallet");
+				}
+
+				return true;
+			}
+			catch (ArgumentNullException ex)
```

Comment:
`Exception` instead of `ArgumentNullException`, and move this to the last place after the `NullReferenceException` catching.
The `InvalidOperationException` above would not be caught anywhere, but with this change, both will be.

Reply:
Yep, you are right. Thank you, I forgot to do that.